cluster.c

```diff
@@ -697,31 +684,33 @@ void ShardingInfoRDBLoad(RedisModuleIO *rdb)
 
     /* Load individual shard groups */
     for (int i = 0; i < rdb_shard_groups_num; i++) {
-        ShardGroup sg;
-        ShardGroupInit(&sg);
+        ShardGroup *sg = ShardGroupCreate();
 
         size_t len;
         char *buf = RedisModule_LoadStringBuffer(rdb, &len);
-        RedisModule_Assert(len < sizeof(sg.id));
-        memcpy(sg.id, buf, len);
-        sg.id[len] = '\0';
+        RedisModule_Assert(len < sizeof(sg->id));
+        memcpy(sg->id, buf, len);
+        sg->id[len] = '\0';
         RedisModule_Free(buf);
+        if (RedisModule_LoadUnsigned(rdb)) {
+            sg->local = true;
+        }
```

Comment:
```suggestion
        sg->local = RedisModule_LoadUnsigned(rdb);
```

Would this be easier to understand?

Reply:
well, sort of.  implicit cast, but you're probably right.